ambry-commons/src/test/java/com.github.ambry.commons/NettyByteBufLeakHelper.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.commons;
+
+import io.netty.buffer.PoolArenaMetric;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.buffer.PooledByteBufAllocatorMetric;
+import java.util.List;
+import org.junit.Assert;
+
+
+/**
+ * A helper class to detect Netty ByteBuf leak. To use this class, create a {@link NettyByteBufLeakHelper} in your test
+ * class and call {@link NettyByteBufLeakHelper#beforeTest()} at any method that has {@link org.junit.Before} tag and
+ * call {@link NettyByteBufLeakHelper#afterTest()} at any method that has {@link org.junit.After} tag.
+ */
+public class NettyByteBufLeakHelper {
+  private long activeHeapAllocations;
+  private long activeDirectAllocations;
+
+  private long heapAllocations;
+  private long heapDeallocations;
+
+  private long directAllocations;
+  private long directDeallocations;
+
+  // when the cache is enable, just don't check the allocation and deallocation.
+  private boolean cachedEnabled;
+
+  /**
+   * Constructor to create a {@link NettyByteBufLeakHelper}.
+   */
+  public NettyByteBufLeakHelper() {
+    cachedEnabled = Integer.parseInt(System.getProperty("io.netty.alllocator.maxCachedBufferCapacity", "0")) != 0;
```

Comment:
This should be `allocator` instead of `alllocator`.
Does netty have a constant for the property name that can be used here?

Reply:
good catch, It doesn't really have a constant for those property keys. Will update.